src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -0,0 +1,185 @@
+                    return;
+
+                throw new LicenseLimitException(LimitType.PullReplicationAsHub, "Your license doesn't support adding Hub Replication feature.");
+
+            case nameof(UpdateExternalReplicationCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasExternalReplication)
+                    if (serverStore.LicenseManager.LicenseStatus.HasDelayedExternalReplication)
+                        return;
+
+                if (databaseRecord.ExternalReplications.Last().DelayReplicationFor == TimeSpan.Zero)
+                    throw new LicenseLimitException(LimitType.ExternalReplication, "Your license doesn't support adding External Replication feature.");
+
+                throw new LicenseLimitException(LimitType.DelayedExternalReplication, "Your license doesn't support adding Delayed External Replication.");
+
+            case nameof(AddRavenEtlCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasRavenEtl)
+                    return;
+
+                throw new LicenseLimitException(LimitType.RavenEtl, "Your license doesn't support adding Raven ETL feature.");
+
+            case nameof(AddSqlEtlCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasSqlEtl)
+                    return;
+
+                throw new LicenseLimitException(LimitType.SqlEtl, "Your license doesn't support adding SQL ETL feature.");
+
+            case nameof(EditTimeSeriesConfigurationCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasTimeSeriesRollupsAndRetention)
+                    return;
+
+                if (databaseRecord.TimeSeries.Collections.Count > 0 && databaseRecord.TimeSeries.Collections.Last().Value.Disabled == false)
+                    throw new LicenseLimitException(LimitType.TimeSeriesRollupsAndRetention, "Your license doesn't support adding Time Series Rollups And Retention feature.");
+
+                return;
+
+            case nameof(EditDocumentsCompressionCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasDocumentsCompression)
+                    return;
+
+                throw new LicenseLimitException(LimitType.DocumentsCompression, "Your license doesn't support adding Documents Compression feature.");
```

Comment:
Let's check if the configuration is disabled. If yes, we can skip it.

Reply:
Done